apply_factors_wrangler.py

```diff
@@ -0,0 +1,222 @@
+
+def get_from_sqs(queue_url):
+    """
+    Gets and formats data from sqs queue
+    :param queue_url - String: Name of sqs queue
+    :return: factors_dataframe - Dataframe: Data from previous step
+             receipt_handle    - String: The id of the sqs message, used to delete from queue at the end of the process
+    """
+
+    response = receive_message(queue_url)
+    message = response['Messages'][0]
+    message_json = json.loads(message['Body'])
+    factors_dataframe = pd.DataFrame(message_json)
+    # Used for clearing the Queue
+    receipt_handle = message['ReceiptHandle']
+
+    return factors_dataframe, receipt_handle
+
+
+def receive_message(queue_url):
+    """
+    Gets the raw sqs response.
+    :param queue_url - String: Name of sqs queue
+    :return: response - Dict: unprocessed(raw) sqs message.
+    """
+
+    sqs = boto3.client('sqs', region_name='eu-west-2')
+    return sqs.receive_message(QueueUrl=queue_url)
+
+def delete_sqs_message(sqs, queue_url, receipt_handle):
+    """
+    Deletes a specified sqs message from the queue
+    :param sqs - Boto3 client(SQS): SQS client for use in interacting with sqs
+    :param queue_url - String: Name of sqs queue
+    :param receipt_handle    - String: The id of the sqs message, used to delete from queue
+    :return: None
+    """
+    sqs.delete_message(QueueUrl=queue_url, ReceiptHandle=receipt_handle)
+
+
+def send_output_to_sqs(queue_url, message, output_message_id,receipt_handle):
+    """
+        Handles sending output data at end of module, and deleting input data from queue
+        :param queue_url - String: Name of sqs queue
+        :param message - String: Message to send to sqs, string(json) representation of output dataframe
+        :param output_message_id  - String: ID of the message to be sent
+        :param receipt_handle    - String: The id of the sqs message, used to delete from queue
+        :return: None
+    """
+    sqs = boto3.client('sqs', region_name='eu-west-2')
+    send_sqs_message(sqs, queue_url, message, output_message_id)
+    delete_sqs_message(sqs, queue_url, receipt_handle)
+
+def send_sqs_message(sqs, queue_url, message, output_message_id):
+    """
+    This method is responsible for sending data to the SQS queue.
+    :param sqs - Boto3 client(SQS): SQS client for use in interacting with sqs
+    :param queue_url: The url of the SQS queue. - Type: String.
+    :param message: The message/data you wish to send to the SQS queue - Type: String.
+    :param output_message_id: The label of the record in the SQS queue - Type: String
+    :return: None
+    """
+
+    sqs.send_message(QueueUrl=queue_url,
+                     MessageBody=message,
+                     MessageGroupId=output_message_id,
+                     MessageDeduplicationId=str(random.getrandbits(128))
+                     )
+
+
+def lambda_handler(event, context):
+    try:
+
+        # Set up clients
+        # # S3
+        bucket_name = os.environ['bucket_name']
```

Comment:
Need to do the same input validation steps we'd mentioned recently....

`os.environ['some_value']` == bad
`os.environ.get('some_value, None)` + handling None == better
using `marshmallow` to validate your inputs  == happy days